bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -48,15 +50,15 @@
  *
  *
  */
-class PendingAddOp implements WriteCallback, TimerTask {
+class PendingAddOp implements WriteCallback, TimerTask, IntProcedure {
     private final static Logger LOG = LoggerFactory.getLogger(PendingAddOp.class);
 
     ByteBuf toSend;
     AddCallback cb;
     Object ctx;
     long entryId;
     int entryLength;
-    Set<Integer> writeSet;
+    IntHashSet writeSet;
```

Comment:
Technically I think we don't need Set<Integer> here. I bet an `int[]` would work here as well.